book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -600,23 +600,61 @@ order on byte strings is the numeric ordering).
 
 We use the following rocksdb column families:
 
-| Tree                 |                  Keys |                              Values |
-|----------------------|-----------------------|-------------------------------------|
-| `hash_by_height`     | `BE32(height)`        | `block::Hash`                       |
-| `height_by_hash`     | `block::Hash`         | `BE32(height)`                      |
-| `block_by_height`    | `BE32(height)`        | `Block`                             |
-| `tx_by_hash`         | `transaction::Hash`   | `(BE32(height) \|\| BE32(tx_index))`|
-| `utxo_by_outpoint`   | `OutPoint`            | `TransparentOutput`                 |
-| `sprout_nullifiers`  | `sprout::Nullifier`   | `()`                                |
-| `sapling_nullifiers` | `sapling::Nullifier`  | `()`                                |
-| `orchard_nullifiers` | `orchard::Nullifier`  | `()`                                |
-| `sprout_anchors`     | `sprout::tree::Root`  | `()`                                |
-| `sapling_anchors`    | `sapling::tree::Root` | `()`                                |
+| Column Family         | Keys                  | Values                               | Updates |
+|-----------------------|-----------------------|--------------------------------------|---------|
+| `hash_by_height`      | `BE32(height)`        | `block::Hash`                        | Never   |
+| `height_by_hash`      | `block::Hash`         | `BE32(height)`                       | Never   |
+| `block_by_height`     | `BE32(height)`        | `Block`                              | Never   |
+| `tx_by_hash`          | `transaction::Hash`   | `(BE32(height) \|\| BE32(tx_index))` | Never   |
+| `utxo_by_outpoint`    | `OutPoint`            | `transparent::Output`                | Delete  |
+| `sprout_nullifiers`   | `sprout::Nullifier`   | `()`                                 | Never   |
+| `sapling_nullifiers`  | `sapling::Nullifier`  | `()`                                 | Never   |
+| `orchard_nullifiers`  | `orchard::Nullifier`  | `()`                                 | Never   |
+| `sprout_anchors`      | `sprout::tree::Root`  | `()`                                 | Never   |
+| `sprout_incremental`  | `BE32(height)`        | `sprout::tree::NoteCommitmentTree`   | Delete  |
+| `sapling_anchors`     | `sapling::tree::Root` | `()`                                 | Never   |
+| `sapling_incremental` | `BE32(height)`        | `sapling::tree::NoteCommitmentTree`  | Delete  |
+| `orchard_anchors`     | `orchard::tree::Root` | `()`                                 | Never   |
+| `orchard_incremental` | `BE32(height)`        | `orchard::tree::NoteCommitmentTree`  | Delete  |
```

Comment:
The `_incremental` may need to change to be a "global" non-keyed value (not sure exactly how), see comment below.

Reply:
We have a few options here:
* make the Key `()` - a placeholder for an empty value, and update the serialized tree for every block
* make the key the tip height, but serialize the whole tree at that height, then delete the previous height, for every block

Either way works, and they will look identical outside the finalized state. The `height` key is slightly easier to debug, the `()` key might be slightly easier to program or understand.

```suggestion
| `sprout_incremental`  | `BE32(height)` *?*    | `sprout::tree::NoteCommitmentTree`   | Delete  |
| `sapling_anchors`     | `sapling::tree::Root` | `()`                                 | Never   |
| `sapling_incremental` | `BE32(height)` *?*    | `sapling::tree::NoteCommitmentTree`  | Delete  |
| `orchard_anchors`     | `orchard::tree::Root` | `()`                                 | Never   |
| `orchard_incremental` | `BE32(height)` *?*    | `orchard::tree::NoteCommitmentTree`  | Delete  |
```